src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2809,3 +2809,45 @@ void Creature::ReleaseFocus(Spell const* focusSpell)
     if (focusSpell->GetSpellInfo()->HasAttribute(SPELL_ATTR5_DONT_TURN_DURING_CAST))
         ClearUnitState(UNIT_STATE_ROTATING);
 }
+
+float Creature::GetAttackDistance(Unit const* player) const
+{
+    float aggroRate = sWorld->getRate(RATE_CREATURE_AGGRO);
+
+    if (aggroRate == 0)
+        return 0.0f;
+
+    if (!player)
+        return 0.0f;
+
+    uint32 playerLevel = player->getLevelForTarget(this);
+    uint32 creatureLevel = getLevelForTarget(player);
+
+    int32 levelDiff = int32(playerLevel) - int32(creatureLevel);
```

Comment:
Sorry, I definitely missed these. I added the suggested changes to make the pull easier for you.

```suggestion
    int32 levelDiff = static_cast<int32>(playerLevel) - static_cast<int32>(creatureLevel);
```

Reply:
cool because i was going to merge it soon :D